### DISCUSSION TITLE
Run Query Option Above SOQL Queries with Results in SOQL Builder Table View

### DISCUSSION THREAD

**Discussion by rlancejohnson** [forcedotcom/salesforcedx-vscode#5859] [Ideas]
**Is your feature request related to a problem? Please describe.**
-  I have to always toggle from the Text Editor to the SOQL Builder just to run the query.
- I have to highlight a long query in an apex method to get a sample of it's output.
- Queries with more than a few columns render terribly in the terminal.

**Describe the solution you'd like**
As a developer, I want the ability to run a query with a simple Run Query option that hovers above the query (similar to the Run Test option above @isTest methods) from any file with a .tgr, .cls, .apex, or .soql extension with the results opened in the table view of the SOQL Query Builder, so that I can easily understand the results of the query and run queries more easily on the fly. This would be a significant improvement over the Execute SOQL Query with Currently Selected Text cli command.

**Describe alternatives you've considered**
The Execute SOQL Query with Currently Selected Text cli command is currently available, but terribly inconvenient and isn't available with the new table view of the SOQL Query Builder.

**Additional context**
The Run Test option for @isTest apex methods
![Screenshot 2024-05-31 at 7 02 35 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/36022919/67a5ab3e-1757-4a74-b684-1a4baf7830b7)

The Execute Apex option for anonymous apex files
![Screenshot 2024-05-31 at 7 19 23 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/36022919/f078acdb-86b7-46f0-ac13-dd672fc041c2)

[a comment by an automated account is not shown]

**Comment by git2gus** [comment]
This issue has been linked to a new work item: W-15918420

**Comment by rlancejohnson** [comment]
Any update on this?

**Comment by daphne-sfdc** [comment]
Hi @rlancejohnson,

We recently did add code lens for "Run Query" and "Get Query Plan" at the top of `.soql` files!  Clicking them prints the result into the Output Tab and saves a CSV, same as the behavior of **SFDX: Execute SOQL Query with Currently Selected Text**.

PR here: https://github.com/forcedotcom/salesforcedx-vscode/pull/7089 🙂
